lxd/db/images.mapper.go

```diff
@@ -65,56 +65,38 @@ func (c *ClusterTx) GetImages(filter ImageFilter) ([]Image, error) {
 	// Result slice.
 	objects := make([]Image, 0)
 
-	// Check which filter criteria are active.
-	criteria := map[string]interface{}{}
-	if filter.Project != "" {
-		criteria["Project"] = filter.Project
-	}
-	if filter.Fingerprint != "" {
-		criteria["Fingerprint"] = filter.Fingerprint
-	}
-	if filter.Public != false {
-		criteria["Public"] = filter.Public
-	}
-	if filter.Cached != false {
-		criteria["Cached"] = filter.Cached
-	}
-	if filter.AutoUpdate != false {
-		criteria["AutoUpdate"] = filter.AutoUpdate
-	}
-
 	// Pick the prepared statement and arguments to use based on active criteria.
 	var stmt *sql.Stmt
 	var args []interface{}
 
-	if criteria["Project"] != nil && criteria["Public"] != nil {
+	if filter.Project != nil && filter.Public != nil {
```

Comment:
@masnax what if I provide Project and Public filters, but *also* provide Autoupdate or Cached too? They seem to be being ignored. This seems like an (admittedly existing) bug.